homeassistant/components/leneda/config_flow.py

```diff
@@ -0,0 +1,678 @@
+"""Config flow for Leneda integration."""
+
+from __future__ import annotations
+
+from asyncio import Task
+from collections.abc import Mapping
+from datetime import datetime, timedelta
+import logging
+from typing import Any, Final
+
+from leneda import LenedaClient
+from leneda.exceptions import ForbiddenException, UnauthorizedException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlowResult
+from homeassistant.core import callback
+from homeassistant.helpers import selector
+
+from .const import (
+    CONF_API_TOKEN,
+    CONF_ENERGY_ID,
+    CONF_METERING_POINTS,
+    DOMAIN,
+    SENSOR_TYPES,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+# Setup types
+SETUP_TYPE_PROBE: Final = "probe"
+SETUP_TYPE_MANUAL: Final = "manual"
+
+# Error messages
+ERROR_INVALID_METERING_POINT: Final = "invalid_metering_point"
+ERROR_SELECT_AT_LEAST_ONE: Final = "select_at_least_one"
+ERROR_UNKNOWN: Final = "unknown"
+ERROR_EXISTING_CONFIG_NOT_FOUND: Final = "existing_config_not_found"
+ERROR_UPDATED_EXISTING: Final = "updated_existing"
+ERROR_DUPLICATE_METERING_POINT: Final = "duplicate_metering_point"
+ERROR_FORBIDDEN: Final = "forbidden"
+ERROR_UNAUTHORIZED: Final = "unauthorized"
+ERROR_NO_OBIS_CODES: Final = "no_obis_codes"
+
+
+class LenedaConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Leneda."""
+
+    VERSION = 1
```

Comment:
Version defaults to `1` anyway so can be removed.

Reply:
True. But is there any downside to being explicit here? I feel it would also be a good reminder about potential migrations when the need arises.